module/zfs/zfs_rlock.c

```diff
@@ -457,18 +467,20 @@ zfs_rangelock_enter_reader(zfs_rangelock_t *rl, zfs_locked_range_t *new)
 	 * locks and bumping ref counts (r_count).
 	 */
 	zfs_rangelock_add_reader(tree, new, prev, where);
+	return (B_TRUE);
 }
 
 /*
  * Lock a range (offset, length) as either shared (RL_READER) or exclusive
  * (RL_WRITER or RL_APPEND).  If RL_APPEND is specified, rl_cb() will convert
  * it to a RL_WRITER lock (with the offset at the end of the file).  Returns
  * the range lock structure for later unlocking (or reduce range if the
- * entire file is locked as RL_WRITER).
+ * entire file is locked as RL_WRITER), or NULL if nonblock is true and the
+ * lock could not be acquired immediately.
  */
-zfs_locked_range_t *
-zfs_rangelock_enter(zfs_rangelock_t *rl, uint64_t off, uint64_t len,
-    zfs_rangelock_type_t type)
+static zfs_locked_range_t *
+_zfs_rangelock_enter(zfs_rangelock_t *rl, uint64_t off, uint64_t len,
```

Comment:
nit: Rather than using `_` prefix convention, would you mind renaming `_zfs_rangelock_enter()` to `zfs_rangelock_enter_impl()` to follow the convention used in the rest of the OpenZFS code.

Reply:
Certainly, done.